hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -269,6 +284,160 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
     }
   }
 
+  @Override
+  public boolean dropView(TableIdentifier identifier) {
+    if (!isValidIdentifier(identifier)) {
+      return false;
+    }
+
+    try {
+      String database = identifier.namespace().level(0);
+      String viewName = identifier.name();
+      Table table = clients.run(client -> client.getTable(database, viewName));
+      HiveViewOperations.validateTableIsIcebergView(
+          table, CatalogUtil.fullTableName(name, identifier));
+      clients.run(
+          client -> {
+            client.dropTable(database, viewName);
+            return null;
+          });
+      LOG.info("Dropped View: {}", identifier);
+      return true;
+
+    } catch (NoSuchViewException | NoSuchObjectException e) {
+      LOG.info("Skipping drop, View does not exist: {}", identifier, e);
+      return false;
+    } catch (TException e) {
+      throw new RuntimeException("Failed to drop " + identifier, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to dropView", e);
+    }
+  }
+
+  @Override
+  public List<TableIdentifier> listViews(Namespace namespace) {
+    Preconditions.checkArgument(
+        isValidateNamespace(namespace), "Missing database in namespace: %s", namespace);
+
+    try {
+      return listTablesByType(
+          namespace, TableType.VIRTUAL_VIEW, BaseMetastoreTableOperations.ICEBERG_VIEW_TYPE_VALUE);
+    } catch (UnknownDBException e) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to list all views under namespace " + namespace, e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to listViews", e);
+    }
+  }
+
+  private List<TableIdentifier> listTablesByType(
+      Namespace namespace, TableType tableType, String tableTypeProp)
+      throws TException, InterruptedException {
+    String database = namespace.level(0);
+    List<String> tableNames = clients.run(client -> client.getTables(database, "*", tableType));
+
+    // Retrieving the Table objects from HMS in batches to avoid OOM
+    List<TableIdentifier> filteredTableIdentifiers = Lists.newArrayList();
+    Iterable<List<String>> tableNameSets = Iterables.partition(tableNames, 100);
+
+    for (List<String> tableNameSet : tableNameSets) {
+      filteredTableIdentifiers.addAll(filterIcebergTables(tableNameSet, namespace, tableTypeProp));
+    }
+
+    return filteredTableIdentifiers;
+  }
+
+  private List<TableIdentifier> filterIcebergTables(
```

Comment:
Would be nice to make the listTables() method call this?

Reply:
+1